piqe_ocp_lib/api/resources/ocp_nodes.py

```diff
@@ -343,6 +343,53 @@ def mark_node_unschedulable(self, node_name: str) -> Optional[ResourceInstance]:
                 logger.error("Exception encountered while marking node unschedulable: %s\n", e)
         return api_response
 
+    def make_node_cordon(self, node_name: str) -> bool:
+        """
+        Make node cordon and return whether it's cordon or not
+        :param node_name:
+        :return:
+        """
+        api_response = None
+        body = {
+            "spec": {
+                "taints": [{"effect": "NoSchedule", "key": "node.kubernetes.io/unschedulable"}],
+                "unschedulable": True,
+            }
+        }
+        try:
+            api_response = self.ocp_nodes.patch(name=node_name, body=body)
+            logger.info("Node %s maked cordon" % node_name)
+        except ApiException as e:
+            logger.error("Exception encountered while making node cordon: %s\n", e)
+        if "'unschedulable': True" in str(api_response.spec):
+            made_cordon = True
+        else:
+            made_cordon = False
+        return made_cordon
+
+    def make_node_uncordon(self, node_name: str) -> bool:
+        """
+        Make node uncordon and return whether it's uncordon or not
+        :param node_name:
+        :return:
+        """
+        body = {
```

Comment:
I don't think it's sufficient to just set 'unschedulable' to False. We should be removing the whole 'taints' element from 'spec'. We can do this by passing ``` "op": "remove", "path": "....```
Please refer to the following:
https://stackoverflow.com/questions/27439986/what-is-the-json-patch-format-to-remove-an-element-from-an-array

Reply:
okay, I do agree, thanks!

Reply:
For this jsonpatch needs to be installed explicitely.